drivers/timer/arcv2_timer0.c

```diff
@@ -137,10 +145,26 @@ static void timer_int_handler(void *unused)
 	/* clear the interrupt by writing 0 to IP bit of the control register */
 	timer0_control_register_set(_ARC_V2_TMR_CTRL_NH | _ARC_V2_TMR_CTRL_IE);
 
+#ifdef CONFIG_SMP
+	u64_t curr_time;
+	k_spinlock_key_t key;
+
+	/* gfrc is the wall clock */
+	curr_time = arc_connect_gfrc_read();
+
+	key = k_spin_lock(&lock);
```

Comment:
So that same code is supposed to be executed by all cores in the cluster?
If so it might be a bit of overkill I guess, but if only 1 core will be doing that then spinlock is not required.

Reply:
Accoridng to my debug on hsdk,  it's possible for all cores to raise local timer interrupt, so spinlock is required to get the wall time

Currently, Zephyr's SMP is still simple.  In the future,  it's possible to optimize this

Reply:
Sure that's perfectly possible but do we really need slave cores to have Timer0 interrupt for anything except timekeeping? If not we may disable it. Note since we do have cross-core interrupts I guess it's OK for slave cores to not have their own periodic interrupt.

Reply:
I think Alexey has a point, what if multiple CPUs announce the same clocktick? I think then the first one does the 'real' announce, and next ones do a z_clock_announce(0) ? I think z_clock_announce can handle that, but it seems a waste of cycles.